pkg/flow/tracer_ringbuf.go

```diff
@@ -91,15 +91,9 @@ func (m *RingBufTracer) listenAndForwardRingBuffer(debugging bool, forwardCh cha
 	if debugging {
 		m.stats.logRingBufferFlows(mapFullError)
 	}
-	// if the flow was received due to lack of space in the eBPF map
```

Comment:
wouldn't removing this logic means we will stay in hmap full condition now for longer and possibly dropping flows ?

Reply:
I agree that having a fixed-rate flush could be a better solution, but actually there's already a flush every 5s (the `cacheActiveTimeout` setting) ; users can set it to 1s if that works better for them

Reply:
or we introduce a new "stress timer" (could be automatically set to `cacheActiveTimeout / something`) that only starts when stress is detected (ie. maps are full) ? but that's a bit complex, it has to be worth it

Reply:
by the way: when you said "this logic means we will stay in hmap full condition now for longer and possibly dropping flows" I don't think this is true; when maps are full, flows are not dropped, they're moved to the ring buffer. Dropped flows are only for busy maps when there isn't a new flow creation, so not something that can lead the map being full (we're updating an entry in that case, not adding a new one, so no incidence on hmap size)

Reply:
we use rb for new flows and it send 1 pkt at a time to userspace so it can't handle burst of traffic either

Reply:
it still handle that much better than an uncontrolled hot-loop of flush events. Look at the -50% CPU that I'm showing in the PR description

Reply:
I'll run perf-scale tests to check the diff

Reply:
/LGTM if perf scale shows improvements

Reply:
ingress-perf shows slightly improved stats: -4% memory, -8% CPU
(https://docs.google.com/spreadsheets/d/1EN12dogz-0_H_5tSoV24T4iHa9a3wSgoF396YyOqavA/edit?gid=93930639#gid=93930639 / diff: https://docs.google.com/spreadsheets/d/1q_XCJ48h2Q78JapxgNB37nJe-4lTYVlMx7xVrn7ztck/edit?gid=696044201#gid=696044201)

Now running cluster-density. But I'm not sure to see something very different here, as the RB isn't involved a lot anyway.

Problem is that, none of those test really stress the agents. So I may not see something better than this -8% here. The tests that I did above DO stress the agents much more, and I had -50%

Reply:
@msherif1234 , @Amoghrd did run cluster-density, with also slightly positive results: https://issues.redhat.com/browse/NETOBSERV-1564?focusedId=24995474&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-24995474